pkg/reconciler/pipelinerun/pipelinerun_updatestatus_test.go

```diff
@@ -0,0 +1,670 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelinerun
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+	logtesting "knative.dev/pkg/logging/testing"
+)
+
+type updateStatusTaskRunsData struct {
+	withConditions map[string]*v1beta1.PipelineRunTaskRunStatus
+	missingTaskRun map[string]*v1beta1.PipelineRunTaskRunStatus
+	foundTaskRun   map[string]*v1beta1.PipelineRunTaskRunStatus
+	recovered      map[string]*v1beta1.PipelineRunTaskRunStatus
+	simple         map[string]*v1beta1.PipelineRunTaskRunStatus
+}
+
+func getUpdateStatusTaskRunsData() updateStatusTaskRunsData {
```

Comment:
I should note that this as a separate function/struct makes more sense if you see the [tests added in my WIP branch for the implementation](https://github.com/abayer/tektoncd-pipeline/blob/embedded-statuses-impl-mk2/pkg/reconciler/pipelinerun/pipelinerun_updatestatus_test.go#L1159) - there, we want to be able to test the function that calls either `updatePipelineRunStatusFromTaskRuns` and `updatePipelineRunStatusFromRuns` _or_ the added-in-the-impl `updatePipelineRunStatusFromChildRefs` (or all of 'em, for `embedded-status=both`). It didn't make sense to me to copy-paste the structs I end up using in both the existing test and the new one when I could just generate them in a function both the tests can call. I went with a struct as the return from that function rather than a map just because I thought it'd be more clear, but I'm not married to that choice. =)